hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/TargetRepository.java

```diff
@@ -223,6 +223,17 @@ void setAssignedAndInstalledDistributionSetAndUpdateStatus(@Param("status") Targ
     @Query("SELECT t FROM JpaTarget t WHERE t.id IN ?1")
     List<JpaTarget> findAllById(Iterable<Long> ids);
 
+    /**
+     * Finds all targets for the given list of controller IDs.
+     * 
+     * @param controllerIds
+     *            The controller IDs to look for.
+     * 
+     * @return The list of matching targets.
+     */
+    @Query("SELECT t FROM JpaTarget t WHERE t.controllerId IN ?1")
+    List<JpaTarget> findAllByControllerId(Iterable<String> controllerIds);
```

Comment:
check if partitioning of controllerIds is done in calling code

Reply:
yep, the calling code partitions the list of controller IDs and makes multiple calls (i.e. multiple SQL queries) to above method if needed